OrbitCore/CallstackData.cpp

```diff
@@ -20,8 +25,88 @@ void CallstackData::AddUniqueCallStack(CallStack call_stack) {
   unique_callstacks_[hash] = std::make_shared<CallStack>(std::move(call_stack));
 }
 
-void CallstackData::AddCallStackFromKnownCallstackData(
-    const orbit_client_protos::CallstackEvent& event, const CallstackData* known_callstack_data) {
+uint32_t CallstackData::GetCallstackEventsCount() const {
+  uint32_t count = 0;
+  for (const auto& tid_and_events : callstack_events_by_tid_) {
+    count += tid_and_events.second.size();
+  }
+  return count;
+}
+
+absl::flat_hash_map<int32_t, uint32_t> CallstackData::GetCallstackEventsCountsPerTid() const {
+  absl::flat_hash_map<int32_t, uint32_t> counts;
+  for (const auto& tid_and_events : callstack_events_by_tid_) {
+    counts.emplace(tid_and_events.first, tid_and_events.second.size());
+  }
+  return counts;
+}
+
+uint32_t CallstackData::GetCallstackEventsOfTidCount(int32_t thread_id) const {
+  if (thread_id == SamplingProfiler::kAllThreadsFakeTid) {
```

Comment:
As said in a different place. I am not sure, whether we should keep the allthreads data separate or just have it in the same map. looks like it making stuff slightly more complicated then needed.

Reply:
This is outdated now.